modules/internal/ChapelAutoAggregation.chpl

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright 2020-2021 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module ChapelAutoAggregation {
+  private use CommAggregation;
+
+  config param verboseAggregation = false;
+  config param yieldFrequency = 1024;
+  config param dstBuffSize = 4096;
+  config param srcBuffSize = 4096;
```

Comment:
For tuning for different networks it is really useful to be able to set these at execution time. I was thinking they could be `private config const`, but that still shows up in our `--help` output. Maybe they could just be `private const` but have them take their value from an env var still?

Reply:
Interesting. I didn't like either option (have `--help` polluted vs have to use env), but will go for the environment variable approach.

Maybe, down the road, we can think of ways to add `config const`s only if some `config param` or compiler flag was used. In other words, it would be great to have these show up in the `--help` output if the user compiled with `--auto-aggregation`.

Reply:
It seems to me that this problem is connected to how `--help` works. If we had in here, say, `private config const dstBuffSize = 4096` and then the only way to set it was using the module name, e.g. `./myProgram --ChapelAutoAggregation.dstBuffSize=8192`, then I wouldn't expect `dstBuffSize` to appear in the `--help` output, unless you explicitly asked for `--help` output for `ChapelAutoAggregation` somehow.

It seems to me that we should make an issue about this.

Reply:
https://github.com/Cray/chapel-private/issues/1605